### DISCUSSION TITLE
Any ways to display controls/elements on the top of the NativeControlHost?

### DISCUSSION THREAD

**Discussion by laolarou726** [AvaloniaUI/Avalonia#13460] [Q&A]
Currently, NativeControlHost will always be placed at the top of all other controls in the UI. Any way to "merge" the native control with other Avalonia original controls? I'm using the [LibVLCSharp.Avalonia](https://code.videolan.org/videolan/LibVLCSharp) package to display a video.

**Comment by dme-compunet** [comment]
@laolarou726 I think it's only possible with Popup or Flyout

**Comment by dme-compunet** [comment]
By the way, if you are in windows only, then there is also the option of receiving mouse events on `NativeControlHost` by setting the window as transparent (with `WS_EX_TRANSPARENT`) and then Windows routes the events to the window below it.

**Reply by laolarou726** [reply]
Thank you so much for your help. I was trying to build a basic video viewer with some texts and controls on it. It seems currently the Avalonia is still pretty limited on the media control stuffs.

**Reply by thevortexcloud** [reply]
There is a few third party video controls. The biggest one I know of is the VLC one, which is officially supported by VLC.

https://github.com/videolan/libvlcsharp

In any case, I am pretty sure this is the airspace problem. WPF has the same problem.

Also on X11 it's possible to make windows click through. But it does require using an optional X11 extension that is not particularly well documented.
